### PR TITLE
notifications: adds library notifications

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds new notifications: booking, request and transit notice.
* Adds a communication language for the libraries notifications.
* Adds comments and clean up the notifications creation code.
* Uses a class property for notification types instead of a string in
  the code.
* Enhance some formatted strings.
* Adds new notification templates in 4 languages.
* Removes old location based notifications (paging request).
* Removes useless debug printing instructions.
* Adds a new function to convert an ISO639-2 into an ISO639-1 language
  code using the iso639 python package.
* Removes useless cli command.
* Removed the useless pycountry package dependency.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Johnny Mariéthoz <johnny.mariethoz@rero.ch>

## Dependencies

My PR depends PR(s):
- https://github.com/rero/rero-ils-ui/pull/606

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1229?milestone=291300

## How to test?

- ! This replace an old implementation: (paging request)

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — CHANGES_REQUESTED** [review]
A verb is missing in the commit message title.

File: rero_ils/modules/loans/api.py [hunk 1]
```diff
@@ -562,34 +564,69 @@ def create_notification(self, notification_type=None, counter=0):
...
+            return self._send_notification(record)
+
+        if (
+            # transit_notice
+            notif_type == Notification.TRANSIT_NOTICE_NOTIFICATION_TYPE
+            # booking
+            or notif_type == Notification.BOOKING_NOTIFICATION_TYPE
+            # request
+            or notif_type == Notification.REQUEST_NOTIFICATION_TYPE
+        ):
```
Comment (lauren-d) [inline rero_ils/modules/loans/api.py:612]: Could be replaced by :
```python
if notif_type in [
           Notification.TRANSIT_NOTICE_NOTIFICATION_TYPE,
           Notification.BOOKING_NOTIFICATION_TYPE,
           Notification.REQUEST_NOTIFICATION_TYPE]:
```

**Review by lauren-d — APPROVED** [review]

**Review by rerowep — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]

**Comment by ManaDeweerdt** [comment]
I'm not sure if it is linked or already reported but the 'destination' is missed from the transit message.
image

**Comment by Garfield-fr** [comment]
> I'm not sure if it is linked or already reported but the 'destination' is missing from the transit message.
> image

The text is not injected, because the translation is missing

**Comment by ManaDeweerdt** [comment]
I don't understand very well the ''Removes old location based notifications (paging request).'' The field 'send notification' was still there at the location level.

**Comment by jma** [comment]
> I don't understand very well the ''Removes old location based notifications (paging request).'' The field 'send notification' was still there at the location level.

We has an old paging request implementation. With this new US, I have re-implement this feature as it is  one of the 3 new notifications required by this new US. My comment was: please retest this feature as it has been rewritten.

**Pull request merged**